TST: numpy.VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences is deprecated

<!-- This comments are hidden when you submit the issue,
so you do not need to remove them! -->

<!-- Please be sure to check out our contributing guidelines,
https://github.com/astropy/astropy/blob/master/CONTRIBUTING.md .
Please be sure to check out our code of conduct,
https://github.com/astropy/astropy/blob/master/CODE_OF_CONDUCT.md . -->

<!-- Please have a search on our GitHub repository to see if a similar
issue has already been posted.
If a similar issue is closed, have a quick look to see if you are satisfied
by the resolution.
If not please go ahead and open an issue! -->

<!-- Please check that the development version still produces the same bug.
You can install development version with
pip install git+https://github.com/astropy/astropy
command. -->

### Description
<!-- Provide a general description of the bug. -->

`numpy-dev` job on Travis CI is failing due to deprecation warning in `modeling` tests. That job is allowed to fail.

### Example log
<!-- Even if you do not think this is necessary, it is useful information for the maintainers.
Please run the following snippet and paste the output below:
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import numpy; print("Numpy", numpy.__version__)
import astropy; print("astropy", astropy.__version__)
import scipy; print("Scipy", scipy.__version__)
-->

https://travis-ci.org/github/astropy/astropy/jobs/677857013

```
==================================== ERRORS ====================================
_ ERROR collecting .tox/py37-test-devdeps/lib/python3.7/site-packages/astropy/modeling/tests/test_units_mapping.py _
../../.tox/py37-test-devdeps/lib/python3.7/site-packages/astropy/modeling/tests/test_units_mapping.py:51: in <module>
    @pytest.mark.parametrize(("model", "input_units", "return_units"), _models_with_units())
../../.tox/py37-test-devdeps/lib/python3.7/site-packages/astropy/modeling/tests/test_units_mapping.py:48: in _models_with_units
    return np.array([models, input_units, return_units]).T
E   numpy.VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray
_ ERROR collecting .tox/py37-test-devdeps/lib/python3.7/site-packages/astropy/modeling/tests/test_units_mapping.py _
../../.tox/py37-test-devdeps/lib/python3.7/site-packages/astropy/modeling/tests/test_units_mapping.py:51: in <module>
    @pytest.mark.parametrize(("model", "input_units", "return_units"), _models_with_units())
../../.tox/py37-test-devdeps/lib/python3.7/site-packages/astropy/modeling/tests/test_units_mapping.py:48: in _models_with_units
    return np.array([models, input_units, return_units]).T
E   numpy.VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray
!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!
======================== 4 skipped, 2 errors in 23.96s =========================
ERROR: InvocationError for command /home/travis/build/astropy/astropy/.tox/py37-test-devdeps/bin/pytest --pyargs astropy /home/travis/build/astropy/astropy/docs --remote-data=any (exited with code 2)
```

--- a/astropy/modeling/tests/test_units_mapping.py
+++ b/astropy/modeling/tests/test_units_mapping.py
@@ -3,7 +3,6 @@ import pytest
 import numpy as np
 
 from astropy import units as u
-from astropy.units import Quantity, UnitsError, equivalencies
 from astropy.modeling.core import Model, fix_inputs
 from astropy.modeling.models import Polynomial1D
 
@@ -11,7 +10,7 @@ from astropy.modeling.models import Polynomial1D
 class _ExampleModel(Model):
     n_inputs = 1
     n_outputs = 1
-    
+
     def __init__(self):
         self._input_units = {"x": u.m}
         self._return_units = {"y": u.m/u.s}
@@ -45,7 +44,7 @@ def _models_with_units():
                     {'y0': u.Unit("m / s"), 'y1': u.Unit("m / s")},
                     {'y0': u.Unit("m / s"), 'y1': u.Unit("m / s")}
                     ]
-    return np.array([models, input_units, return_units]).T
+    return np.array([models, input_units, return_units], dtype=object).T
 
 
 @pytest.mark.parametrize(("model", "input_units", "return_units"), _models_with_units())

